### PR TITLE
Add CSS variable transformer for sd

### PR DESCRIPTION
**Pull request by douglaszaltron**
#56

### CODE REVIEW

**Comment by lukasoppermann** [comment]
Hey @douglaszaltron, 

I somehow figured this would be part of a `formatter`?
Why do you favor a transformer?

It has the benefit of flexibility but would need to be applied in correct order. Also did you test it with nested tokens. We must make sure to avoid issues where the referenced token is already transformed to a css value (this could be done using the `original` values I think).

Somehow it feels more fitting for me to have this done in the formatter, I guess in [this function](https://github.com/lukasoppermann/style-dictionary-utils/blob/main/src/utilities/jsonToNestedValue.ts#L12). But I am open to being convinced that transformers are better.

**Comment by douglaszaltron** [comment]
> Hey @douglaszaltron,
> 
> I somehow figured this would be part of a `formatter`? Why do you favor a transformer?
> 
> It has the benefit of flexibility but would need to be applied in correct order. Also did you test it with nested tokens. We must make sure to avoid issues where the referenced token is already transformed to a css value (this could be done using the `original` values I think).
> 
> Somehow it feels more fitting for me to have this done in the formatter, I guess in [this function](https://github.com/lukasoppermann/style-dictionary-utils/blob/main/src/utilities/jsonToNestedValue.ts#L12). But I am open to being convinced that transformers are better.

At first, I thought about creating it as a formater, but as a transformer, I can apply the value modification for both JavaScript/* and JSON/*. It would extend more easily. I did realize about the order issue, that the value mutations need to be done beforehand.

```
transforms: [
    "name/cti/kebab",
    "shadow/css",
    "fontFamily/css",
    "font/css",
    "variables/css",
],
```

File: src/transformer/variables-css.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * variablesCss
+ * @description convert the `name` to a css variable
+ */
+export const variablesCss: StyleDictionary.Transform = {
+  type: `value`,
+  transitive: true,
+  transformer: (token: StyleDictionary.TransformedToken, options: StyleDictionary.Options) => {
+    const cssVarPrefix = getCssVarPrefix(options);
+    const varPrefix = cssVarPrefix ? `${cssVarPrefix}-` : ''
+    return `var(--${varPrefix}${token.name})`
```
Comment (douglaszaltron) [inline src/transformer/variables-css.ts:21]: I believe it's better to add the value as fallback.

`var(--${varPrefix}${token.name}, token.value)`

Comment (lukasoppermann) [reply src/transformer/variables-css.ts:21]: Maybe this can be an option? e.g. `valueFallback` ?

Comment (douglaszaltron) [reply src/transformer/variables-css.ts]: @lukasoppermann I made the adjustments, take a look.

**Comment by douglaszaltron** [comment]
@lukasoppermann I made the adjustments, take a look.

**Comment by lukasoppermann** [comment]
Looks great, I'll have to test it and will merge it afterwards.

**Pull request merged**